ludwig/data/preprocessing.py

```diff
@@ -124,11 +121,40 @@ def build_metadata(dataset_df, features, global_preprocessing_parameters):
             preprocessing_parameters = global_preprocessing_parameters[
                 feature[TYPE]
             ]
+
+        # deal with encoders that have fixed preprocessing
+        if 'encoder' in feature:
+            encoders_registry = get_from_registry(
+                feature[TYPE],
+                input_type_registry
+            ).encoder_registry
+            encoder_class = encoders_registry[feature['encoder']]
+            if hasattr(encoder_class, 'fixed_preprocessing_parameters'):
+                encoder_fpp = encoder_class.fixed_preprocessing_parameters
+
+                if 'preprocessing' in feature:
```

Comment:
This block I don't like, I thought i remved it but probaby popped up in the merge. it's fine to keep it for now, but will add a task to remove it later.